build/run-in-docker

```diff
@@ -55,18 +57,24 @@ if (( $# == 0 )); then
   DOCKER_OPTS="${DOCKER_OPTS} -it"
   RUN_CMD="/bin/bash"
 else
-  RUN_CMD="${@@Q}"
+  THIS_FILE=$(basename $SCRIPTPATH)
+  ARG0_CMD=$(basename "$0")
```

Comment:
Not sure I'm a fan of this hack, as I could easily see it causing as many problems as it solves once a user forgets they've setup this aliasing.  But as-is, it's inconsistent.  If someone wants to run an argless command this way, it doesn't work because the logic is missing in the `$# == 0` case.

It's also a bit odd that the headline implies this is just about allowing a variable to be overridden, but the majority of the change in this PR are actually for this unrelated change.

Reply:
Thanks for pointing out the bug with  `$# == 0`.  I do not really need the symlink based dispatch for this PR